### PR TITLE
Use downloader from installer library

### PR DESCRIPTION
**Pull request by shorim**
**Description**

Changes proposed in this pull request:
- remove `deploy`, `git` and `download` packages
- use `git` package from `parallel-install` library to clone Kyma repo
- use `download` package from `parallel-install` library to download files
- improve doc for workspace flag

**Related issue(s)**

#809

### CODE REVIEW

**Review by clebs — APPROVED** [review]

File: docs/gen-docs/kyma_alpha_deploy.md [hunk 1]
```diff
@@ -33,7 +33,8 @@ kyma alpha deploy [flags]
       --value strings             Set a configuration value (e.g. --value component.key='the value').
   -f, --values-file strings       Path to a JSON or YAML file with configuration values.
       --workers-count int         Number of parallel workers used for the deployment. (default 4)
-  -w, --workspace string          Path used to download Kyma sources. (default "workspace")
+  -w, --workspace string          If --source is local, then workspace should be a path already containing Kyma sources.
+                                  If --source is not local, then workspace would be the path used to download Kyma sources. (default "workspace")
```
Comment (NHingerl) [inline docs/gen-docs/kyma_alpha_deploy.md:37]:
```suggestion
                                  If --source is not local, then workspace must be the path used to download Kyma sources (default: "workspace").
```

**Review by NHingerl — COMMENTED** [review]
minor language editing

File: docs/gen-docs/kyma_alpha_deploy.md (hunk 1 above)
Comment (shorim) [reply docs/gen-docs/kyma_alpha_deploy.md:37]: the part for (default "workspace") is auto-generated and we should not change it manually

Comment (shorim) [reply docs/gen-docs/kyma_alpha_deploy.md:37]: I will change "would" to "must" as suggested, but the other changes should not be applied

**Review by clebs — APPROVED** [review]

File: docs/gen-docs/kyma_alpha_deploy.md (hunk 1 above)
Comment (NHingerl) [reply docs/gen-docs/kyma_alpha_deploy.md]: Alright! I didn't know that this is auto-generated. 
I'm not a fan of how it is rendered tho. It's better with a colon and in front of the full stop, but no need to change it here.

File: docs/gen-docs/kyma_alpha_deploy.md [hunk 2]
```diff
@@ -33,7 +33,8 @@ kyma alpha deploy [flags]
       --value strings             Set a configuration value (e.g. --value component.key='the value').
   -f, --values-file strings       Path to a JSON or YAML file with configuration values.
       --workers-count int         Number of parallel workers used for the deployment. (default 4)
-  -w, --workspace string          Path used to download Kyma sources. (default "workspace")
+  -w, --workspace string          If --source is local, then workspace should be a path already containing Kyma sources.
+                                  If --source is not local, then workspace must be the path used to download Kyma sources. (default "workspace")
```
Comment (majakurcius) [inline docs/gen-docs/kyma_alpha_deploy.md:37]:
```suggestion
  -w, --workspace string          If --source is local, then workspace is the path already containing Kyma sources.
                                  If --source is not local, then workspace is the path used to download Kyma sources. (default "workspace")
```

File: cmd/kyma/alpha/deploy/cmd.go [hunk 3]
```diff
@@ -52,7 +54,8 @@ func NewCmd(o *Options) *cobra.Command {
 		Aliases: []string{"d"},
 	}
 
-	cobraCmd.Flags().StringVarP(&o.WorkspacePath, "workspace", "w", defaultWorkspacePath, "Path used to download Kyma sources.")
+	cobraCmd.Flags().StringVarP(&o.WorkspacePath, "workspace", "w", defaultWorkspacePath,
+		"If --source is local, then workspace should be a path already containing Kyma sources.\nIf --source is not local, then workspace must be the path used to download Kyma sources.")
```
Comment (majakurcius) [inline cmd/kyma/alpha/deploy/cmd.go:58]:
```suggestion
		"If --source is local, then workspace is the path already containing Kyma sources.\nIf --source is not local, then workspace is the path used to download Kyma sources.")
```

**Review by majakurcius — APPROVED** [review]

**Review by clebs — APPROVED** [review]

**Review by clebs — APPROVED** [review]

**Review by majakurcius — APPROVED** [review]

**Pull request merged**